zooniverse/Panoptes-Front-End
# Pages Editor: implement Associated Tutorials

## PR Overview

Part of: **Pages Editor MVP** project and **FEM Lab** super-project
Follows #7048
Staging branch URL: https://pr-???.pfe-preview.zooniverse.org/lab/1982/workflows/editor/3711?env=staging

This PR implements the "Associated Tutorials" feature on the Workflow Settings sub-page.

- New component: AssociatedTutorials
- New behaviour/feature: on Pages Editor (for any Workflow) -> Workflow Settings...
  - The _Associated Tutorials_ feature should list all Tutorials available to the Project _(caveat: up to a max of 100.)_
  - The Associated Tutorials feature should show which Tutorial is linked to the Workflow via radio buttons. There's an option for "no tutorial".
  - Clicking on the radio button should link/unlink the Tutorial to/from the Workflow.

Linking Tutorials was a bit weird.
- Unlike Subject Sets, a Workflow should have either 0 or 1 Tutorial. It shouldn't have multiple; not sure how our PFE or FEM Classifiers would even handle 2+ tutorials.
- Unlike Subject Sets, a Workflow's linked tutorial(s) aren't listed in `workflow1234.links`. Instead, we have to query `/api/tutorials?workflow_id=1234`

_Screenshot: example workflow with 2 available tutorials, and the first selected as this wf's tutorial._

<img width="303" alt="image" src="https://github.com/zooniverse/Panoptes-Front-End/assets/13952701/bc63922e-f6a8-4cad-b761-776f0037b08e">

### Testing

- Open the staging URL. You should be looking at a test workflow (e.g. staging wf 3711)
- Click on the Workflow Settings tab/button to view the sub-page.
- Look for the Associated Tutorials feature.
  - It should list all available tutorials, and show which one (if any) is selected as the wf's tutorial.
  - Click on a tutorial to set it.
  - Reload the page to confirm.

### Status

Ready for review!

This PR is currently targeting pages-editor-pt15 for review purposes. Do not merge until 7048 is ready and this branch's merge is re-targeted to `master`.
